packages/driver/src/cy/ensures.ts

```diff
@@ -383,6 +384,32 @@ export const create = (state: StateFunc, expect: $Cy['expect']) => {
     })
   }
 
+  /**
+   * ensureCommandCanCommunicateWithAUT will check if the command should be able to communicate with the AUT
+   * If we can not communicate, throw an error.
+   * Intended to use within retry loops.
+   * err: optional error to pass end to be appended to if the assertion happened while the aut was cross origin.
+   * @returns true or throws an error
+   */
+  const ensureCommandCanCommunicateWithAUT = (err?): boolean => {
+    if (!isRunnerAbleToCommunicateWithAut()) {
+      const crossOriginCommandError = $errUtils.errByPath('miscellaneous.cross_origin_command', {
+        commandOrigin: window.location.origin,
+        autOrigin: Cypress.state('autLocation').originPolicy,
+      })
+
+      if (err) {
+        err.message = `${err.message}\n\n${crossOriginCommandError.message}`
```

Comment:
```suggestion
        err.message = $errUtils.appendErrMsg(err, crossOriginCommandError.message)
```

Just mutating the message will leave the stack (which includes the message) out of sync. You'll need to import `$errUtils` above too of course.